csp-upgrade-insecure-requests/index.html

```diff
@@ -0,0 +1,102 @@
+limitations under the License.
+-->
+<html lang="en">
+  <head>
+    <meta charset="utf-8">
+
+    <!-- // [START meta-tag] -->
+    <meta http-equiv="Content-Security-Policy" content="upgrade-insecure-requests">
+    <!-- // [END meta-tag] -->
+
+    <meta http-equiv="X-UA-Compatible" content="IE=edge">
+
+    <meta name="description" content="Sample illustrating the use of Content-Security-Policy: upgrade-insecure-requests.">
+
+    <meta name="viewport" content="width=device-width, initial-scale=1">
+
+    <title>Upgrade Insecure Resource Requests Sample</title>
+
+    <!-- Add to homescreen for Chrome on Android -->
+    <meta name="mobile-web-app-capable" content="yes">
+    <link rel="icon" sizes="192x192" href="../images/touch/chrome-touch-icon-192x192.png">
+
+    <!-- Add to homescreen for Safari on iOS -->
+    <meta name="apple-mobile-web-app-title" content="Upgrade Insecure Resource Requests Sample">
+
+    <meta name="apple-mobile-web-app-capable" content="yes">
+    <meta name="apple-mobile-web-app-status-bar-style" content="black">
+    <link rel="apple-touch-icon-precomposed" href="../images/apple-touch-icon-precomposed.png">
+
+    <!-- Tile icon for Win8 (144x144 + tile color) -->
+    <meta name="msapplication-TileImage" content="images/touch/ms-touch-icon-144x144-precomposed.png">
+    <meta name="msapplication-TileColor" content="#3372DF">
+
+    <link rel="icon" href="../images/favicon.ico">
+
+    <link rel="stylesheet" href="../styles/main.css">
+  </head>
+
+  <body>
+    <h1>Upgrade Insecure Resource Requests Sample</h1>
+
+    <p>Available in <a href="http://www.chromestatus.com/feature/6534575509471232">Chrome 43+</a></p>
+
+    <p>
+      The "<a href="https://w3c.github.io/webappsec/specs/upgrade/">Upgrade Insecure Requests</a>"
+      <a href="http://www.html5rocks.com/en/tutorials/security/content-security-policy/">Content Security Policy</a>
+      can be used to automatically upgrade insecure (e.g. <code>http:</code>) resource requests to
+      a secure alternative (e.g. <code>https:</code>) before a browser fetches them.
+    </p>
+
+    <p>
+      In practice, this helps avoid mixed-content warnings when a page is accessed via
+      <code>https:</code>, but it contains references to resources using absolute
+      <code>http:</code> URLs.
+    </p>
+
+    <p>
+      Like other Content Security Policies, the recommend approach is to enable it via a HTTP
+      response header, <code>Content-Security-Policy: upgrade-insecure-requests</code>. However,
+      if you do not have control over the underlying web server (as is the case in this demo), an
+      <a href="http://www.html5rocks.com/en/tutorials/security/content-security-policy/#the-meta-tag">alternative</a>
+      is to include the
+      <code>&lt;meta http-equiv="Content-Security-Policy" content="upgrade-insecure-requests"&gt;</code>
+      tag in your HTML, preferably in the <code>&lt;head&gt;</code>.
```

Comment:
It has to be inside `<head>`, we'll ignore the policy if it appears in the body.

(Not sure you need to go into the detail, but you can inject it into the head via JavaScript; all that matters is that `<head>` is an ancestor of `<meta>`.)